client/src/cli/queryParams.ts

```diff
@@ -1,42 +1,83 @@
 import path from 'path';
-import { getProvider, readJsonFromFile, saveJsonToFile } from "./utils";
+import { getRpcUrl, readJsonFromFile, saveJsonToFile } from "@axiom-crypto/circuit/cliHandler/utils";
 import { buildSendQuery } from "../sendQuery";
 import { argsArrToObj } from '../axiom/utils';
+import {
+  getAxiomV2QueryAddress,
+  getAxiomV2QueryBlockhashOracleAddress,
+  getAxiomV2QueryBroadcasterAddress,
+} from '../lib';
+import { ChainConfig } from '../types';
 
 export const queryParams = async (
   callbackTarget: string,
   options: {
     refundAddress: string;
     sourceChainId: string;
-    callbackExtraData: string;
-    caller: string;
-    argsMap: boolean;
+    targetChainId?: string;
```

Comment:
I think it's safer to have a single optional type which carries all the cross-chain configuration parameters.  Right now it's hard to check the code because many of the optional flags carry redundant info.

Reply:
I do agree that it would be a lot easier to have the optional type that carries the crosschain params, but it doesn't seem like there's a super straightforward way for commander to take in an object from the CLI (since all CLI options are mapped 1:1 with this `options` input object)... i can investigate a bit more tomorrow to see if there's any potential workaround

Reply:
Maybe we can have an explicit part of the function which checks the presence of these flags is consistent and sets an isCrosschain flag that's used after.